### PR TITLE
Cluster-autoscaler: test cloud provider for unit tests

### PR DESCRIPTION
**Pull request by mwielgus**
Ref: #2191

cc: @fgrzadkowski @piosz @jszczepkowski

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/test/test_cloud_provider.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+	sync.Mutex
+	nodes      map[string]string
+	groups     map[string]cloudprovider.NodeGroup
+	onIncrease func(string, int) error
+	onDelete   func(string, string) error
+}
+
+// OnIncreaseFunc is a function called on node group increase in TestCloudProvider.
+// First parameter is the NodeGroup id, second is the increase delta.
+type OnIncreaseFunc func(string, int) error
```
Comment (jszczepkowski) [inline cluster-autoscaler/cloudprovider/test/test_cloud_provider.go]: Can you add names to the arguments?

File: cluster-autoscaler/cloudprovider/test/test_cloud_provider.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+// OnDeleteFunc is a function called on cluster
+type OnDeleteFunc func(string, string) error
```
Comment (jszczepkowski) [inline cluster-autoscaler/cloudprovider/test/test_cloud_provider.go]: Can you add names to the arguments?

File: cluster-autoscaler/cloudprovider/test/test_cloud_provider.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+	for _, group := range tcp.groups {
+		result = append(result, group)
+	}
+	return result
+}
+
+// NodeGroupForNode returns the node group for the given node, nil if the node
+// should not be processed by cluster autoscaler, or non-nil error if such
+// occurred.
+func (tcp *TestCloudProvider) NodeGroupForNode(node *apiv1.Node) (cloudprovider.NodeGroup, error) {
```
Comment (jszczepkowski) [inline cluster-autoscaler/cloudprovider/test/test_cloud_provider.go]: This function currently never returns error, so, I'm not sure if it is reasonable to keep error in the signature.

**Comment by jszczepkowski** [comment]
In general LGTM, just some nits.

File: cluster-autoscaler/cloudprovider/test/test_cloud_provider.go (hunk 2 above)
Comment (mwielgus) [inline cluster-autoscaler/cloudprovider/test/test_cloud_provider.go]: same as above.

**Pull request merged**